### PR TITLE
Upgrade to terraform 0.12

### LINKED ISSUE #51: Upgrade to support Terraform 0.12
Hi has anybody looked at upgrading the provider to support Terraform 0.12?

### PR DESCRIPTION
**Pull request by gertjvr**
Started on upgrading the provider to support 0.12.2 this would resolve #51.

All the tests pass few things I am unsure about, will comment on the changes.

### CODE REVIEW

File: teamcity/resource_feature_commit_status_publisher.go [hunk 1]
```diff
@@ -162,7 +162,7 @@ func buildGithubCommitStatusPublisher(d *schema.ResourceData) (api.BuildFeature,
 		opt = api.NewCommitStatusPublisherGithubOptionsPassword(host, local["username"].(string), local["password"].(string))
 	}
 
-	return api.NewFeatureCommitStatusPublisherGithub(opt)
+	return api.NewFeatureCommitStatusPublisherGithub(opt, "")
```
Comment (gertjvr) [inline teamcity/resource_feature_commit_status_publisher.go]: Is the assumption correct to apply this to all vcsroots.

File: teamcity/resource_vcs_root_git.go [hunk 2]
```diff
@@ -388,10 +388,9 @@ func expandGitVcsRootOptions(d *schema.ResourceData) (*api.GitVcsRootOptions, er
 		return nil, err
 	}
 
-	_, authSpecified := d.GetOkExists("auth")
-	if authSpecified {
+	if a, ok := d.GetOk("auth"); ok {
```
Comment (gertjvr) [inline teamcity/resource_vcs_root_git.go]: weird issue, d.GetOkExists("auth") always returns true even when "auth" wasn't set.

**Comment by cvbarros** [comment]
Amazing @gertjvr ! At this moment I'm travelling, but next week I'll circle back on this for tests/review and I think there won't be any blockers for merging it!

**Comment by nrccua-grante** [comment]
this looks like a very useful change and will help with getting Team City implemented into TF.  
+1

**Comment by jspenc72** [comment]
@cvbarros Please merge this asap if this is ready to go.

**Comment by jspenc72** [comment]
@cvbarros If you need another reviewer for these pull requests I am happy to assist.

**Comment by gertjvr** [comment]
Octopus just upgraded there provider to [v0.12](https://github.com/OctopusDeploy/terraform-provider-octopusdeploy/pull/32), my urgency to use this has just gone up, @cvbarros can you please provide a timeframe when this could be reviewed / merged.

**Comment by jspenc72** [comment]
@cvbarros I need this as well.

**Comment by cvbarros** [comment]
Pulling this branch now to review / test. Then if no issues, should be merged soon and I'll make a new release

**Comment by cvbarros** [comment]
All gtg, thanks a bunch @gertjvr! Sorry for the delay in getting it reviewed.

**Comment by jspenc72** [comment]
👍👍👍👍 @cvbarros

**Comment by jspenc72** [comment]
No worries at all, thanks for maintaining this. Please let us know if we can help.

**Pull request merged**